### DISCUSSION TITLE
Cannot bind to Control+Enter or Shift+Enter for chat-like behavior in TextArea

### DISCUSSION THREAD

**Discussion by espositocode** [Textualize/textual#4216] [Q&A] [answered]
Based on these set of keys, I do not see a binding for Control+Enter or Control+Shift (by itself): https://github.com/Textualize/textual/blob/main/src/textual/keys.py 

Is there a technical limitation that prevents these keys from used? 

This binding would be particularly useful for TextArea, where the user can distinguish between "submit" and "newline". For example, in most chat apps you can press one of these sequences to generate a newline, where normally pressing enter would immediately send the text.

If I were to build a chat app with Textual, how would I mimic this behavior?

**Comment by davep** [comment, marked as answer]
You may find [this FAQ entry](https://textual.textualize.io/FAQ/#why-do-some-key-combinations-never-make-it-to-my-app) helpful here.

**Reply by espositocode** [reply]
Thanks @davep. So I tried that script to print out the key combinations, and it only prints `enter` every time. 

Just to confirm, this is because the Terminal app doesn't pass along `shift` or `control` with enter, and there is basically nothing Textual can do to work around this?

**Reply by davep** [reply]
Indeed. Here's kitty, iTerm, wezterm, tabby, Alacritty and Contour, each using `showkey`, where I'm trying the three key combinations you mention:

![Screenshot 2024-02-26 at 14 00 40](https://github.com/Textualize/textual/assets/28237/0b14f4b0-0581-467f-aedb-e631b1dd6b3c)

**Reply by davep** [reply]
You may also find #3440 interesting too.

**Reply by espositocode** [reply]
Ok. Thanks for the help
